cache/blobs/compression.go

```diff
@@ -102,21 +102,30 @@ func detectCompressionType(cr io.Reader) (CompressionType, error) {
 
 // GetMediaTypeForLayers retrieves media type for layer from ref information.
 func GetMediaTypeForLayers(diffPairs []DiffPair, ref cache.ImmutableRef) []string {
```

Comment:
Add a comment that on error, slice with partial data would be returned. This confused me in the beginning.